### PR TITLE
Updates to Floquet drive, two-level qubit, and Kerr nonlinear oscillator, as well as some single-charge stuff

### PR DESCRIPTION
**Pull request by shoumikdc**
All these qubits are now updated to work with the standard truncation scheme in qcsys

### CODE REVIEW

**Comment by shoumikdc** [comment]
Summary: 
- Updated transmon qubit to have a gate-offset charge `ng`. By default `ng = 0` to ensure backward compatibility with sims where `ng` was not specified. 
- Added single-electron charge basis mode to the transmon qubit. 
- Updated Floquet drive to use `jnp2jqt` during operator construction.
- Added Floquet tutorial.
- Minor fixes to Kerr nonlinear oscillator (KNO) and IdealQubit.

**Review by Phionx — CHANGES_REQUESTED** [review]
Mostly LGTM! Just need to address some minor comments. Also, run all the tutorial notebooks to confirm everything still works as before.

File: qcsys/devices/drive.py [hunk 1]
```diff
@@ -31,20 +32,20 @@ def label(self):
     def ops(self):
         return self.common_ops()
 
-    def common_ops(self) -> Dict[str, Array]:
+    def common_ops(self) -> Dict[str, jqt.Qarray]:
         ops = {}
 
         M_max = self.M_max
```
[the remark this answers is not in the document]
Comment (shoumikdc) [reply qcsys/devices/drive.py:38]: This is done in the `6-floquet` tutorial and in some of my other simulations! Checked that the drive indeed works with `jit` (and also `vmap` etc).

File: qcsys/devices/kno.py [hunk 2]
```diff
@@ -15,10 +16,6 @@ class KNO(Device):
     Kerr Nonlinear Oscillator Device.
     """
 
-    @classmethod
```
[the remark this answers is not in the document]
Comment (shoumikdc) [reply qcsys/devices/kno.py:18]: Added!

File: qcsys/devices/ideal_qubit.py [hunk 3]
```diff
@@ -17,7 +17,7 @@ class IdealQubit(Device):
     """
 
     def common_ops(self):
-        """ Written in the linear basis. """
```
[the remark this answers is not in the document]
Comment (shoumikdc) [reply qcsys/devices/ideal_qubit.py:20]: Added!

**Comment by shoumikdc** [comment]
Added params validation for KNO and IdealQubit and checked that drive plays nicely with jit in functions that construct driven systems. 

Checked that all previous tutorials run and give expected results!
Since the transmon qubit had the most significant changes, also did a deeper dive into differences between the old and new versions of the Hamiltonian construction. When printed to 8 decimal places in the notebook, the eigenvalues look the same - so the new version matches up. I will note that when I looked at the result to more sig figs, I found that the Hamiltonians are slightly different on the order of 1e-14 (around numerical precision), giving rise to differences in eigenvalues between the old and new versions also on the order of 1e-14. This is mostly expected behavior for 64bit floats and hopefully should not affect any prior results.

**Review by Phionx — APPROVED** [review]
LGTM!

**Pull request merged**